mo_net/train/trainer/trainer.py

```
@@ -301,44 +306,36 @@ def _training_loop(self) -> TrainingResult:
-                match self._post_epoch(L_val):
-                    case CheckFailed() as check:
-                        return TrainingFailed(
-                            model_checkpoint_path=self._model_checkpoint_path,
-                            message=check.message,
-                            model_checkpoint_save_epoch=self._L_val_min_epoch,
-                        )
-                    case None:
-                        pass
-                    case never:
-                        assert_never(never)
+                    self._model.dump(open(self._model_checkpoint_path, "wb"))
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)